### PR TITLE
fix bug in call to BCEwithlogits

### PR DESCRIPTION
**Pull request by aalok-sathe**
the keyword had a typo: 'weight' instead of 'weights'.
in case weights requires grads, it needs to be detached first as BCEwithlogits is not implemented for grad tracked tensor

### CODE REVIEW

**Comment by fcakyon** [comment]
@aalok-sathe feel free to use this improved/fixed/battle-tested version, includes this fix and much more: https://github.com/fcakyon/balanced-loss#improvements
